plugins/apitracing/tools/FunctionDefinitionsChecker.py

```diff
@@ -0,0 +1,103 @@
+    multiple_parameter_definitions = (struct_names & backing_parameter_names)
+    if multiple_parameter_definitions:
+        print("Parameters which are defined as both a struct and a backing parameter", multiple_parameter_definitions)
+
+
+def find_ring_in_highlevel_parameter_types(highlevel_parameter_node):
+    for address_width, param_types in highlevel_parameter_node.items():
+        for param in param_types.keys():
+            visited_high_level_params = set(param)
+            current_node = param
+            while current_node in param_types.keys():
+                current_node = param_types[current_node]
+                if current_node in visited_high_level_params:
+                    print("Highlevelparameter resolution for address width:", address_width, "starting from:",
+                          param, "contains ring between",
+                          param_types[current_node],
+                          "and", current_node)
+                    break
+                visited_high_level_params.add(current_node)
+
+
+def get_parameter_types(modules_node):
+    parameter_types = set()
+    for module_name, function in modules_node.items():
+        for function_name, parameters in function.items():
+            if "Parameters" in parameters.keys():
+                for param_type in parameters["Parameters"].values():
+                    parameter_types.add(param_type)
+    return parameter_types
+
+
+def get_struct_parameter_types(struct_names, structures_node):
+    struct_parameter_types = set()
+    for struct_name, parameters in structures_node.items():
+        for parameter_name, parameter_description in parameters.items():
+            # Don't add structs as parameters since we already iterate over all structs
+            if parameter_description["type"] not in struct_names:
+                struct_parameter_types.add(parameter_description["type"])
+    return struct_parameter_types
+
+
+def get_high_level_parameter_names_and_types(highlevel_parameter_node):
+    highlevel_parameter_names = set()
+    highlevel_parameter_types = set()
+    for address_width, param_types in highlevel_parameter_node.items():
+        for name, highlevel_param_type in param_types.items():
+            highlevel_parameter_names.add(name)
+            highlevel_parameter_types.add(highlevel_param_type)
+    return highlevel_parameter_names, highlevel_parameter_types
+
+
+def get_backing_parameter_names(backing_parameter_node):
+    backing_parameter_names = set()
+    for name, size in backing_parameter_node.items():
+        backing_parameter_names.add(name)
+    return backing_parameter_names
+
+
+if __name__ == '__main__':
+    with open("../configuration/functiondefinitions/functionDefinitions.yaml", "r") as stream:
+        try:
+            function_definitions_yaml = yaml.safe_load(stream)
+            check_parameter_translation(function_definitions_yaml)
+        except yaml.YAMLError as exc:
+            print(exc)
```

Comment:
The plan is to integrate this into the CI as well, right? If so, you should problable return a non-zero exit code in case anything goes wrong.